Initial Implementation (#1)
marmelab/ra-auth-msal: packages/ra-auth-msal/Readme.md

# ra-auth-msal

> An auth provider for [react-admin](https://github.com/marmelab/react-admin) which handles authentication using the [Microsoft Authentication Library (MSAL)](https://learn.microsoft.com/fr-fr/azure/active-directory/develop/msal-overview).

Reviewer: ```suggestion
An auth provider for [react-admin](https://github.com/marmelab/react-admin) that handles authentication using the [Microsoft Authentication Library (MSAL)](https://learn.microsoft.com/fr-fr/azure/active-directory/develop/msal-overview).
```

> This is useful for instance if you want to use [Azure Active Directory](https://azure.microsoft.com/en-us/services/active-directory/) to authenticate your users.

Reviewer: ```suggestion
This is useful when using [Azure Active Directory](https://azure.microsoft.com/en-us/services/active-directory/) to authenticate your users.
```

    redirectUri: "http://localhost:8080/auth-callback",
    // We need to disable this feature because it is already handled by react-admin, and would otherwise conflict
    navigateToLoginRequestUrl: false,
  },
  cache: {
    cacheLocation: "sessionStorage",
    storeAuthStateInCookie: false,
  },
};
```

```jsx
// in src/App.jsx
import React from 'react';
import { Admin, Resource } from 'react-admin';
import { BrowserRouter } from "react-router-dom";
import { PublicClientApplication } from "@azure/msal-browser";
import { LoginPage, msalAuthProvider } from "ra-auth-msal";
import dataProvider from './dataProvider';
import posts from './posts';
import { msalConfig } from "./authConfig";

const myMSALObj = new PublicClientApplication(msalConfig);

const App = () => {
  const authProvider = msalAuthProvider({
    msalInstance: myMSALObj,
  });

  return (
    <BrowserRouter>
       <Admin
           authProvider={authProvider}
           dataProvider={dataProvider}
           title="Example Admin"
           loginPage={LoginPage}
        >
            <Resource name="posts" {...posts} />
      </Admin>
    </BrowserRouter>
   );
};
export default App;
```

> **Tip:** You need to wrap your `<Admin>` component in a `<BrowserRouter>` for this library to work. Indeed, MSAL uses a hash-based routing strategy when redirecting back to your app, which would be incompatible with a `<HashRouter>`.

Reviewer: ```suggestion
**Tip:** You need to wrap your `<Admin>` component in a `<BrowserRouter>` for this library to work. Indeed, MSAL uses a hash-based routing strategy when redirecting back to your app, which is incompatible with a `<HashRouter>`.
```
